### PR TITLE
fix: add min id length admonition

### PR DESCRIPTION
**Pull request by Mercy811**
# Amplitude Developer Docs PR

## Description

`api2.amplitude.com` have no min length requirements for device IDs and user IDs but http v2 api has mim id length to be 5. Emphasize the need to match the user ID's default length to prevent similar [issue](https://amplitude.atlassian.net/browse/AMP-87192) for customers.

## Change type

- [ ] Doc bug fix. Fixes #[insert issue number]. Amplitude contributors include Jira issue number. 
- [ ] Doc update.
- [ ] New documentation.
- [ ] Non-documentation related fix or update.

# PR checklist:

- [ ] My documentation follows the style guidelines of this project.
- [ ] I previewed my documentation on a local server using `mkdocs serve`.
- [ ] Running `mkdocs serve` didn't generate any failures.
- [ ] I have performed a self-review of my own documentation.

@amplitude-dev-docs

### CODE REVIEW

File: docs/data/sdks/ios-swift/migration.md [hunk 1]
```diff
@@ -101,7 +101,7 @@ The configurations for the new SDK are simpler and more consistent across runtim
 |Customize storage provider. NOT SUPPORTED.|`config.storageProvider`|
 |Set up log level. NOT SUPPORTED.|`config.logLevel`|
 |Customize logger provider. NOT SUPPORTED.|`config.loggerProvider`|
-|Overwrite the minimum length deviceId and userId. NOT SUPPORTED.|`config.minIdLength`|
+|Have no length limit on deviceId or userId.|`config.minIdLength`overwrites the minimum length deviceId and userId.|
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/migration.md:104]:
```suggestion
| `deviceId` and `userId` don't have a minimum length.|`config.minIdLength` overwrites the minimum length `deviceId` and `userId`.|
```

@Mercy811 So is the minimum for Amplitude-Swift 5, which you can override with `config.minIdLength`?

File: includes/sdk-min-id-length.md [hunk 2]
```diff
@@ -0,0 +1,2 @@
+!!!warning "Id length limit"
+    The maintenance SDK uses the old SDK endpoint (`api2.amplitude.com`) which has no length limit on device Id and user Id. But the latest SDK uses HTTP V2 API (`api2.amplitude.com/2/httpapi`) which requires the minimum id length is 5 by default. Set `config.minIdLength` to a smalle value if you set user Id or device Id with length less than 5 when migration.
```
Comment (markzegarelli) [inline includes/sdk-min-id-length.md:2]:
```suggestion
    The maintenance SDK uses an old SDK endpoint (`api2.amplitude.com`) which enforces no length limit for `deviceId` and `userId`. The latest SDK uses Amplitude's HTTP V2 API (`api2.amplitude.com/2/httpapi`) and requires identifiers to be at least 5 characters by default. When you migrate to the new SDK, set `config.minIdLength` to a smaller value if you allowed identifiers with fewer than 5 characters.
```

**Review by markzegarelli — CHANGES_REQUESTED** [review]

File: docs/data/sdks/ios-swift/migration.md (hunk 1 above)
Comment (Mercy811) [reply docs/data/sdks/ios-swift/migration.md:104]: Yes!

File: docs/data/sdks/ios-swift/migration.md [hunk 3]
```diff
@@ -101,7 +101,7 @@ The configurations for the new SDK are simpler and more consistent across runtim
 |Customize storage provider. NOT SUPPORTED.|`config.storageProvider`|
 |Set up log level. NOT SUPPORTED.|`config.logLevel`|
 |Customize logger provider. NOT SUPPORTED.|`config.loggerProvider`|
-|Overwrite the minimum length deviceId and userId. NOT SUPPORTED.|`config.minIdLength`|
+| `deviceId` and `userId` don't have a minimum length.|`config.minIdLength` overwrites the minimum length `deviceId` and `userId`.|
```
Comment (markzegarelli) [inline docs/data/sdks/ios-swift/migration.md:104]:
```suggestion
| `deviceId` and `userId` don't have a minimum length.| Minimum length is 5. `config.minIdLength` overwrites the minimum length  of`deviceId` and `userId`.|
```

**Review by markzegarelli — APPROVED** [review]

**Review by justin-fiedler — APPROVED** [review]
Thanks @Mercy811!

**Pull request merged**